### PR TITLE
[A4] update mongodb and connect-mongo dependencies

### PR DESCRIPTION
**Pull request by haroun**
## Summary

Use emulate-mongo-3-driver

## What are the specific steps to test this change?

1. `npm install && npm test` should be green

## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [x] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [x] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
   "author": "Apostrophe Technologies, Inc.",
   "license": "MIT",
   "dependencies": {
+    "@apostrophecms/emulate-mongo-3-driver": "github:apostrophecms/emulate-mongo-3-driver#pro-5766-emulate-mongo-3",
```
Comment (haroun) [inline package.json:34]: Must be switch to version number once released and before merging

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,8 @@
 There is no change in how the `styles` option is configured.
 * `i18n` module now uses the regular `req.redirect` instead of a direct `res.redirect` to ensure redirection, enabling more possibilities for `@apostrophecms/redirect` module
 * Refactors `AposModal` component with composition api to get rid of duplicated code in `AposFocusMixin` and `AposFocus`.
+* `APOS_MONGODB_LOG_LEVEL` has been removed. According to [mongodb documentation](https://github.com/mongodb/node-mongodb-native/blob/main/etc/notes/CHANGES_5.0.0.md#mongoclientoptionslogger-and-mongoclientoptionsloglevel-removed) "Both the logger and the logLevel options had no effect and have been removed."
+* Update `mongodb` to `6.x` and `connect-mongo` to `5.x`. Add `@apostrophecms/emulate-mongo-3-driver` dependency to keep supporting `mongodb@3.x` queries.
```
Comment (boutell) [inline CHANGELOG.md:12]: * Why do we still need a dependency on `mongodb` at all?
* We're forcing the upgrade, originally we said this would be optional. But I think forcing it will be OK because it looks like you provided good coverage for the 3.x expectations people are likely to have and we are doing it to address compatibility with a supported driver, which has historically been acceptable. Also we did it this way for the 2.x emulator and nobody had a problem with that. So let's go ahead.

File: modules/@apostrophecms/doc-type/index.js [hunk 3]
```diff
@@ -2703,8 +2703,9 @@ module.exports = {
               subquery.limit(undefined);
               subquery.page(undefined);
               subquery.perPage(undefined);
-              const mongo = await subquery.toMongo();
-              const count = await mongo.count();
+              // TODO is toMongo needed here?
+              // const mongo = await subquery.toMongo();
```
Comment (boutell) [inline modules/@apostrophecms/doc-type/index.js:2707]: It is, because of the lateCriteria feature which is important for certain edge cases, and also for consistency with any extensions that are created. More importantly, why did you need to change it out — is a feature missing?

File: modules/@apostrophecms/doc/lib/migrations.js [hunk 4]
```diff
@@ -25,7 +25,7 @@ module.exports = (self) => {
 
     addSetDocumentModesMigration() {
       self.apos.migration.add('set-document-modes', async () => {
-        return self.apos.migration.eachDoc(5, async (doc) => {
```
Comment (boutell) [inline modules/@apostrophecms/doc/lib/migrations.js:28]: Huh, old one.

File: modules/@apostrophecms/express/index.js [hunk 5]
```diff
@@ -554,12 +554,13 @@ module.exports = {
           }
           if (!sessionOptions.store.name) {
             // require from this module's dependencies
-            Store = require('connect-mongo')(expressSession);
+            const MongoStore = require('connect-mongo');
+            sessionOptions.store = MongoStore.create(sessionOptions.store.options);
```
Comment (boutell) [inline modules/@apostrophecms/express/index.js:558]: Good catch.

File: package.json [hunk 6]
```diff
@@ -91,7 +92,7 @@
     "mini-css-extract-plugin": "^1.6.0",
     "minimatch": "^3.0.4",
     "mkdirp": "^0.5.5",
-    "mongodb": "^3.6.6",
+    "mongodb": "^6.5.0",
```
Comment (boutell) [inline package.json:95]: Should probably go away here since the emulator is used?

**Review by boutell — CHANGES_REQUESTED** [review]
Couple things

File: modules/@apostrophecms/db/index.js [hunk 7]
```diff
@@ -96,9 +97,10 @@ module.exports = {
           self.connectionReused = true;
           return;
         }
+        // TODO should be removed since it does nothing according to mongodb documentation
```
Comment (haroun) [inline modules/@apostrophecms/db/index.js:100]: @boutell should I remove this block?

File: modules/@apostrophecms/doc-type/index.js (hunk 3 above)
Comment (boutell) [reply modules/@apostrophecms/doc-type/index.js]: You can't add a reference to the collection to the cursor objects as they are created?

File: modules/@apostrophecms/express/index.js (hunk 5 above)
Comment (boutell) [reply modules/@apostrophecms/express/index.js:558]: Per my notes on the A3 PR this can be factored back out of the if statement I think.

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/express/index.js (hunk 5 above)
Comment (haroun) [reply modules/@apostrophecms/express/index.js:558]: cf https://github.com/apostrophecms/apostrophe/pull/4490#discussion_r1561350790

**Review by boutell — APPROVED** [review]

**Review by boutell — APPROVED** [review]
